### PR TITLE
Upgrade Nuxt to 3.8.2

### LINKED ISSUE #1444: Update to Nuxt 3.8.x
### Is your suggested enhancement related to a problem? Please describe.

We still on version 3.5.x.

I think we can update to 3.8.x to use some improvements.
https://github.com/modrinth/knossos/blob/master/package.json#L32

It can also be better for the stability ^^.

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

https://nuxt.com/blog/v3-8

### PR DESCRIPTION
**Pull request by Mysterious-Dev**
All is in the title.

@Geometrically did you see an error when recreating the patch?

### CODE REVIEW

File: composables/how-ago.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { createFormatter, type Formatter } from '@vintl/how-ago'
-import { IntlController } from '@vintl/vintl/controller'
+import { type IntlController } from '@vintl/vintl/controller'
```
Comment (brawaru) [inline composables/how-ago.ts:2]:
```suggestion
import type { IntlController } from '@vintl/vintl/controller'
```

File: composables/compact-number.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { createFormatter, type Formatter } from '@vintl/compact-number'
-import { IntlController } from '@vintl/vintl/controller'
+import { type IntlController } from '@vintl/vintl/controller'
```
Comment (brawaru) [inline composables/compact-number.ts:2]:
```suggestion
import type { IntlController } from '@vintl/vintl/controller'
```

**Comment by Mysterious-Dev** [comment]
@brawaru I applied your suggestions ^^.

**Comment by Mysterious-Dev** [comment]
I updated nuxt to 3.8.2 ^^.

**Comment by Geometrically** [comment]
This breaks the CF pages build (modrinth is hosted on pages) so we are holding off on this for now.

**Pull request closed without merging**